### PR TITLE
[NET-8946 NET-8947 NET-8948] security: bump go, x/net and envoy versions

### PR DESCRIPTION
**Pull request by dduzgun-security**
### Description ###
Upgrade to use Go 1.21.9. This resolves CVEs
[CVE-2023-45288](https://nvd.nist.gov/vuln/detail/CVE-2023-45288) (http2).

Upgrade to support Envoy 1.28.2. This resolves CVEs
[CVE-2024-27919](https://nvd.nist.gov/vuln/detail/CVE-2024-27919) (http2).

Upgrade to use golang.org/x/net v0.24.0. This resolves CVEs
[CVE-2023-45288](https://nvd.nist.gov/vuln/detail/CVE-2023-45288) (x/net).

### Checklist ###
- [ ] Tests added
- [ ] [CHANGELOG entry added](https://github.com/hashicorp/consul-k8s/blob/main/CONTRIBUTING.md#adding-a-changelog-entry)

### CODE REVIEW

File: charts/consul/Chart.yaml [hunk 1]
```diff
@@ -22,7 +22,7 @@ annotations:
     - name: consul-dataplane
       image: docker.mirror.hashicorp.services/hashicorppreview/consul-dataplane:1.5-dev
     - name: envoy
-      image: envoyproxy/envoy:v1.25.11
+      image: envoyproxy/envoy:v1.28.2
```
Comment (zalimeni) [inline charts/consul/Chart.yaml:25]: We actually don't need to modify this - this is a deprecated value specific to the legacy API gateway, which will go away at some point hopefully soon.

This repo actually doesn't have any need to set the Envoy version explicitly, as it's entirely inherited from `consul-dataplane`.

**Review by zalimeni — COMMENTED** [review]
One revert suggested, but otherwise LGTM!

**Comment by zalimeni** [comment]
We'll also want to backport this n-3, through 1.1.x (these versions correspond to Consul's n-2 and LTS)

File: charts/consul/Chart.yaml (hunk 1 above)
Comment (dduzgun-security) [reply charts/consul/Chart.yaml]: Thanks for the details, that's good to know 👀 I have reverted the value: https://github.com/hashicorp/consul-k8s/pull/3893/commits/cce1e7dbc1660f94cdd568e4282874ce1406e202

File: .changelog/3893.txt [hunk 2]
```diff
@@ -0,0 +1,14 @@
+```release-note:security
+Upgrade to use Go `1.21.9`. This resolves CVE
+[CVE-2023-45288](https://nvd.nist.gov/vuln/detail/CVE-2023-45288) (`http2`).
+```
+
+```release-note:security
+Upgrade to support Envoy `1.28.2`. This resolves CVE
+[CVE-2024-27919](https://nvd.nist.gov/vuln/detail/CVE-2024-27919) (`http2`).
+```
```
Comment (zalimeni) [inline .changelog/3893.txt:9]:
```suggestion
```

**Review by zalimeni — APPROVED** [review]
One edit in changelog then LGTM!

**Comment by zalimeni** [comment]
Rebased on `main` to squash down to single commit, setting auto-merge since we've seen acceptance tests pass

**Pull request merged**